packages/@aws-cdk/aws-lambda-event-sources/test/test.kafka.ts

```diff
@@ -0,0 +1,260 @@
+import { arrayWith, expect, haveResource } from '@aws-cdk/assert';
+import { SecurityGroup, SubnetType, Vpc } from '@aws-cdk/aws-ec2';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as msk from '@aws-cdk/aws-msk';
+import { Secret } from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import { Test } from 'nodeunit';
+import * as sources from '../lib';
+import { TestFunction } from './test-function';
+
+export = {
+  'MSK: sufficiently complex example'(test: Test) {
```

Comment:
Usually we start by testing the opposite. We start with 'default'. This test will only provide the mandatory attributes and assert that they are in the output and the defaults come out as expected.

Then we move to specifying the optional attributes and checking that they are interpreted correctly, and finally into cases like vpc.

Can we organize it that way? Take a look at some other examples in this and other recent packages.

Reply:
I believe that it already is organized in that way. Renamed tests to make it more clear.